views/my_packages.register.dt

```diff
@@ -20,20 +20,8 @@ block body
 
 		form(method="POST", action="#{req.rootDir}register_package")
 			p
-				label(for="kind") Repository type
-				select(name="kind", size="1")
-					- import dubregistry.repositories.repository;
-					- if (supportsRepositoryKind("github"))
-						option(value="github", selected=kind == "github") GitHub project
-					- if (supportsRepositoryKind("bitbucket"))
-						option(value="bitbucket", selected=kind == "bitbucket") Bitbucket project
-					- if (supportsRepositoryKind("gitlab"))
-						option(value="gitlab", selected=kind == "gitlab") GitLab project
-			p
-				label(for="owner") Repository owner
-				input(type="text", name="owner", value=owner)
-			p
-				label(for="project") Repository name
-				input(type="text", name="project", value=project)
+				label(for="project") Repository URL
+				input(type="text", name="url", value=url, pattern="^(https?://)?(www\.)?(github(\.com)?|gitlab(\.com)?|bitbucket(\.org)?)/[^/]+/[^/]+([?#.].*)?$", placeholder="URL or kind/owner/project")
+				span.repoError Please input a valid project URL to a GitHub, GitLab or BitBucket project.
```

Comment:
As suggested by Martin,maybe add a few examples here:

https://github.com/dlang/dub
...

Reply:
I added one example in the placeholder (containing github as it is the most common) because placeholders are the easiest and best way to show a valid input imo, if you enter something else you get 2 other examples for gitlab and bitbucket